### PR TITLE
Improve bootstrap-application generator.

### PR DESCRIPTION
**Pull request by mshima**
`bootstrap-application` is used as a helper to entity as core feature.
Remove `bootstrap-application` dependency on `entity` generator.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/generator-base-entity.cjs [hunk 1]
```diff
@@ -17,23 +17,34 @@
...
+  PREPARING_EACH_ENTITY_FIELD_QUEUE,
+  PREPARING_EACH_ENTITY_RELATIONSHIP_QUEUE,
+  POST_PREPARING_EACH_ENTITY_QUEUE,
+  WRITING_ENTITIES_QUEUE,
+} = QUEUES;
 
 /**
  * This is the base class for a generator that generates entities.
  */
 class JHipsterBaseEntityGenerator extends BaseBlueprintGenerator {
```
Comment (Tcharl) [inline generators/generator-base-entity.cjs:45]: Shouldn't it be `JHipsterBaseEntit**ies**Generator` (and the file name called base-entit**ies**?
Here we have no method that configures a single entity

File: generators/generator-base-entity.cjs [hunk 2]
```diff
@@ -63,131 +114,230 @@ class JHipsterBaseEntityGenerator extends BaseBlueprintGenerator {
...
-        this.debug(`Queueing entity tasks ${CONFIGURING_EACH_ENTITY_PRIORITY}`);
-        const tasks = this.extractTasksFromPriority(CONFIGURING_EACH_ENTITY_PRIORITY, { skip: false });
-        this.getEntitiesToConfigure().forEach(({ entityName, entityStorage, entityConfig }) => {
-          this.debug(`Queueing entity tasks ${CONFIGURING_EACH_ENTITY_PRIORITY} for ${entityName}`);
+  /**
+   * @private
+   * Queue entity tasks.
+   */
+  queueEntityTasks() {
+    this.queueTask({
```
Comment (Tcharl) [inline generators/generator-base-entity.cjs:210]: Maybe one sub-method by step?

File: lib/support/shared-data.cjs [hunk 3]
```diff
@@ -19,14 +19,17 @@
...
+      throw new Error('Storage is required for SharedData');
+    }
     this._configDefaultValues = {};
     this._configChoices = {};
 
     // Backward compatibility sharedData
-    this._configOptions = configOptions || {};
-    this._configOptions.sharedEntities = this._configOptions.sharedEntities || {};
-    this._configOptions.sharedApplication = this._configOptions.sharedApplication || {};
+    this._storage = storage;
```
Comment (Tcharl) [inline lib/support/shared-data.cjs:30]: <3 that new name.

File: utils/field.js [hunk 4]
```diff
@@ -453,6 +408,63 @@ function getEnumValuesWithCustomValues(enumValues) {
   });
 }
 
+function prepareClientFieldForTemplates(entityWithConfig, field, generator) {
+  if (field.fieldValidateRulesPatternAngular === undefined) {
```
Comment (Tcharl) [inline utils/field.js:412]: A sub function by client kind? may help for the split in the future

File: utils/field.js [hunk 5]
```diff
@@ -453,6 +408,63 @@ function getEnumValuesWithCustomValues(enumValues) {
...
+      : field.fieldValidateRulesPattern;
+  }
+}
+
+function prepareServerFieldForTemplates(entityWithConfig, field, generator) {
+  if (field.fieldNameAsDatabaseColumn === undefined) {
+    const fieldNameUnderscored = _.snakeCase(field.fieldName);
+    const jhiFieldNamePrefix = generator.getColumnName(entityWithConfig.jhiPrefix);
+
+    if (isReservedTableName(fieldNameUnderscored, entityWithConfig.prodDatabaseType)) {
```
Comment (Tcharl) [inline utils/field.js:430]: Ideally a specific method by check (could be reused by blueprints I guess)

File: utils/user.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  loadRequiredConfigIntoEntity(user, this.jhipsterConfig);
+  // Fallback to defaults for test cases.
+  loadRequiredConfigIntoEntity(user, this.jhipsterDefaults);
+
+  const oauth2 = user.authenticationType === OAUTH2;
+  const userIdType = oauth2 || user.databaseType !== SQL ? TYPE_STRING : this.getPkType(user.databaseType);
+  const fieldValidateRulesMaxlength = userIdType === TYPE_STRING ? 100 : undefined;
+
+  let idField = user.fields.find(field => field.fieldName === 'id');
```
Comment (Tcharl) [inline utils/user.js:63]: Ideally a method by field to add in the conf (**S**OLID)

**Comment by Tcharl** [comment]
LGTM (but would be better if the remarks could be handled)

File: lib/support/shared-data.cjs (hunk 3 above)
Comment (mshima) [reply lib/support/shared-data.cjs:30]: New name makes much more sense.

File: utils/field.js (hunk 4 above)
Comment (mshima) [reply utils/field.js:412]: Yep, split at v8.

File: generators/generator-base-entity.cjs (hunk 2 above)
Comment (mshima) [reply generators/generator-base-entity.cjs]: I prefer it together. It's easier to see the end result.

File: utils/field.js (hunk 5 above)
Comment (mshima) [reply utils/field.js:430]: jdl validation can be improved for modularity.

Comment (mshima) [reply utils/field.js:430]: We should think about this for v8.

File: utils/user.js (hunk 6 above)
Comment (mshima) [reply utils/user.js]: Used a bit different approach to improve the code.

**Pull request merged**